src/server/GameController.cpp

```diff
@@ -22,33 +22,58 @@ GameController::~GameController() {
     //de-construcstructor
 }
 
-void GameController::update(int client_id, sf::Vector2f client_direction) {
-    // Update players
-    mPlayers[client_id]->setDirection(client_direction);
+void GameController::update() {
+    // Get clients
+    std::vector<int> client_ids = mNetworkingServer->getClientIds();
+
+    // Create new players
+    for (const int client_id : client_ids) {
+        if (!mPlayers[client_id]->isActive()) {
+            createPlayer(client_id);
```

Comment:
why is createPlayer called when it's *not* active. surprising code

Reply:
it sort of makes sense, doesn't make me happy- can we think of a better way? Im ok shipping this if what you just wrote is in the code as a comment or in docs (which we don't have a system for lol). It's definitely hard to understand